graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,273 @@
+import numpy as np
+from ..match import GraphMatch as GMP
+from sklearn.base import BaseEstimator
+import itertools
+
+
+class VNviaSGM(BaseEstimator):
+    """
+    This class implements vertex nomination via seeded graph matching
+
+    Parameters
+    ----------
+    G_1: 2d-array, square
+        Adjacency matrix of `G_1`, the graph where voi is known
+
+    G_2: 2d-array, square
+        Adjacency matrix of `G_2`, the graph where voi is not known
+
+    h: int
+        distance between voi used to create induced subgraph on `G_1`
+
+    ell: int
+        distance from seeds to other verticies to create induced subgraphs on `G_1`
+
+    R: int
+        Number of restarts for soft seeded graph matching algorithm
+
+    Attributes
+    ----------
+    n_seeds_used: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: list
+        List of 2 tuples of the format nomination_list = [(j, p_val)] where p_val is the
+        probability that the voi matches to node j in graph B (sorted by descending
+        probability)
+
+
+    References
+    ----------
+    .. [1] Patsolic, HG, Park, Y, Lyzinski, V, Priebe, CE. Vertex nomination via seeded graph matching. Stat Anal Data
+        Min: The ASA Data Sci Journal. 2020; 13: 229– 244. https://doi.org/10.1002/sam.11454
+
+
+
+    """
+
+    def __init__(self, G_1, G_2, h=1, ell=1, R=100):
+        if type(h) is int and h > 0:
+            self.h = h
+        else:
+            msg = "h must be an integer > 0"
+            raise ValueError(msg)
+        if type(ell) is int and ell > 0:
+            self.ell = ell
+        else:
+            msg = "ell must be an integer > 0"
+            raise ValueError(msg)
+        if type(R) is int and R > 0:
+            self.R = R
+        else:
+            msg = "R must be an integer > 0"
+            raise ValueError(msg)
+
+        if G_1.shape[0] != G_1.shape[1] or G_2.shape[0] != G_2.shape[1]:
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+        else:
+            self.G_1 = G_1
+            self.G_2 = G_2
+
+    def fit(self, X, y=[]):
+        """
+        Fits the model to two graphs
+
+        Parameters
+        ----------
+        X: int
+            Vertex of interest (voi)
+
+        y: list
+            List of length two `[seedsA, seedsB]` where first element is
+        the seeds associated with adjacency matrix G_1
+        and the second element the adjacency matrix associated with G_2, note
+        `len(seedsA)==len(seedsB)` The elements of `seeds_A` and `seeds_B` are
+        vertices which are known to be matched, that is, `seeds_A[i]` is matched
+        to vertex `seeds_B[i]`.
```

Comment:
This description (and other similarly formatted sections) display improperly in the documentation, they show up highlighted grey